src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -605,6 +605,175 @@ def load_cert_validator(namespace):
             setattr(namespace, param_name, read_base_64_file(attr))
     return load_cert_validator
 
+
+def get_network_watcher_from_vm(namespace):
+    from azure.mgmt.compute import ComputeManagementClient
+    from azure.cli.core.commands.arm import parse_resource_id
+
+    compute_client = get_mgmt_service_client(ComputeManagementClient).virtual_machines
+    vm_name = parse_resource_id(namespace.vm)['name']
+    vm = compute_client.get(namespace.resource_group_name, vm_name)
+    namespace.location = vm.location  # pylint: disable=no-member
+    get_network_watcher_from_location()(namespace)
+
+
+def get_network_watcher_from_resource(namespace):
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    from azure.cli.core.commands.arm import parse_resource_id
+
+    resource_client = get_mgmt_service_client(ResourceManagementClient).resources
+    resource = resource_client.get_by_id(namespace.resource, '2016-09-01')
+    namespace.location = resource.location  # pylint: disable=no-member
+    get_network_watcher_from_location(remove=True)(namespace)
+
+def get_network_watcher_from_location(remove=False, watcher_name='watcher_name',
+                                      rg_name='watcher_rg'):
+
+    def _validator(namespace):
+        from azure.mgmt.network import NetworkManagementClient
+        from azure.cli.core.commands.arm import parse_resource_id
+
+        location = namespace.location
+        network_client = get_mgmt_service_client(NetworkManagementClient).network_watchers
+        watcher = next((x for x in network_client.list_all() if x.location == location), None)
```

Comment:
Yep, our generic location handler takes care of this. It does the case insensitive check and converts them from long names to the compact names.